internal/server/event_handlers.go

```diff
@@ -62,7 +62,7 @@ func (s *EventServer) handleEvent() func(w http.ResponseWriter, r *http.Request)
 
 		cleanupMetadata(event)
 
-		ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
+		ctx, cancel := context.WithTimeout(r.Context(), 15*time.Second)
```

Comment:
After reading some a bit more on this I am questioning if this is the right choice. My initial thought was that this context would only be cancelled when the HTTP server is shutdown, but it can also be cancelled by the client. It seems like that type of behavior is not really what we want as NC should just attempt to complete the post.

Reply:
If the client cancels the context then it should also cancel the event posting, the reason for this is that the clients are the Flux controller which do have retry/backoff and they are responsible for retying the notification.